src/popup/pages/SignMessage.vue

```diff
@@ -19,10 +20,12 @@ export default defineComponent({
       try {
         const aeSdk = await getAeSdk();
         const message = route.query.message?.toString();
+        const isHexEncodedMessage = message && route.query.encoding?.toString() === 'hex';
+        const displayMessage = isHexEncodedMessage ? hexToUtf8(message) : message;
         const { host, href } = callbackOrigin.value || {} as any;
 
         await openModal(MODAL_MESSAGE_SIGN, {
-          message,
+          message: displayMessage,
```

Comment:
So, it changes only the message shown to the user? I think Andreas meant to sign string decoded from hex. Like,
```
https://wallet.superhero.com/sign-message?encoding=hex&message=41
```
should return the same as
```
https://wallet.superhero.com/sign-message?message=A
```
As I see, currently the first deeplink will sign exactly "41", not "A".